### DISCUSSION TITLE
API v2

### DISCUSSION THREAD

**Discussion by maciaszczykm** [kubernetes/dashboard#7026] [Ideas]
Discussion about moving towards new API structure.

**Comment by maciaszczykm** [comment]
The next step after moving away from Gulp to Makefile can be [splitting frontend and backend into separate containers](https://github.com/kubernetes/dashboard/issues/6799). This will allow us taking small steps approach into splitting current API into micro services with single API gateway.

**Comment by floreks** [comment]
## Goal
```
TBD
```

## High-level Architecture
```
TBD
```

## Repository Management and Structure
```
TBD
```

## Authentication & Authorization Layer
```
TBD
```

## API (GraphQL, REST, gRPC?)
```
TBD
```

## Next Steps
```
TBD
```
